dds_web/security/auth.py

```diff
@@ -33,36 +33,53 @@
 # FUNCTIONS ############################################################################ FUNCTIONS #
 ####################################################################################################
 
+# Error handler -------------------------------------------------------------------- Error handler #
+
 
 @basic_auth.error_handler
 def auth_error(status):
+    """Handles 401 (Unauthorized) or 403 (Forbidden) for basic authentication."""
     return auth_error_common(status)
 
 
 @auth.error_handler
 def auth_error(status):
+    """Handles 401 (Unauthorized) or 403 (Forbidden) for token authentication."""
     return auth_error_common(status)
 
 
 def auth_error_common(status):
+    """Checks if status code is 401 or 403 and raises appropriate exception."""
     if status == http.HTTPStatus.UNAUTHORIZED:
         raise AuthenticationError()
     elif status == http.HTTPStatus.FORBIDDEN:
         raise AccessDeniedError(message="Insufficient credentials")
 
 
+# User roles -------------------------------------------------------------------------- User roles #
+
+
 @basic_auth.get_user_roles
 def get_user_roles(user):
+    """Gets the role corresponding to the current user when basic authentication is used."""
     return get_user_roles_common(user)
 
 
 @auth.get_user_roles
 def get_user_roles(user):
+    """Gets the role corresponding to the current user when token authentication is used."""
     return get_user_roles_common(user)
 
 
 def get_user_roles_common(user):
-    """Return the users role as saved in the db."""
+    """Get the user role.
+
+    If the user is a Researcher, check if a project has been specified and in that case if the user
+    is set as a owner for that project. If so, the user role is Project Owner. If not, the user role
```

Comment:
If the user has Researcher role and a project, which the user has been set as an owner, has been specified, the user role is returned as Project Owner. Otherwise, it is Researcher.